File: selectionsort.py
def findSmallest(arr):
    smallest = arr[0]
    smallest_in = 0

    for i in range (1, len(arr)):
        if arr[i] < smallest:
            smallest = arr[i]
            smallest_in = i
    return smallest_in

def aufSelectionSort(arr):
    newArr = []
    for i in range (len(arr)):
        smallest = findSmallest(arr)
        newArr.append(arr.pop(smallest))
    return newArr

print(aufSelectionSort([4, 78, 34, 9]))

def findBiggest(arr):
    biggest = arr[0]
    biggest_in = 0

    for i in range (1, len(arr)):
        if arr[i] > biggest:
            biggest = arr[i]
            biggest_in = i
    return biggest_in

def abSelectionSort(arr):
    newArr = []
    for i in range(len(arr)):
        biggest = findBiggest(arr)
        newArr.append(arr.pop(biggest))
    return newArr

print(abSelectionSort([4, 78, 34, 9]))

File: selectionsort.java
import java.util.ArrayList;

public class selectionsort {
    public static void main(String[] args) {
        ArrayList<Integer> l = new ArrayList<>();
        ArrayList<Integer> arr = new ArrayList<>();
        for (int num : new int[]{34, 2, 7, 45}) {
            arr.add(num);
        }
        l = SelectionSort(arr);
        System.out.println(l);
    }
    
    public static ArrayList SelectionSort(ArrayList<Integer> arr) {
        ArrayList<Integer> newArr = new ArrayList<>();
        while (arr.size() > 0) {
            int smallest = findSmallest(arr);
            int smallest_index = findSmallestIndex(arr);
            newArr.add(smallest);
            arr.remove(smallest_index);
        }
        return newArr;
    }

    public static int findSmallest(ArrayList<Integer> arr) {
        int smallest = arr.get(0);
        for (int i = 1; i < arr.size(); i++) {
            if (arr.get(i) < smallest) smallest = arr.get(i);
        }
        return smallest;
    }

    public static int findSmallestIndex(ArrayList<Integer> arr) {
        int smallest = arr.get(0);
        int smallest_index = 0;
        for (int i = 1; i < arr.size(); i++) {
            if (arr.get(i) < smallest) {
                smallest = arr.get(i);
                smallest_index = i;
            }
        }
        return smallest_index;
    }
}
